docs/references/architecture/adr-119-dog-mempool-gossip.md

```diff
@@ -0,0 +1,367 @@
+- libp2p gossipSub protocol
+
+
+### FLOOD - the current transaction gossiping protocol in CometBFT
+
+[Flood](https://github.com/cometbft/cometbft/blob/main/spec/mempool/gossip/flood.md) is the protocol currently used by CometBFT to gossip transactions. If a transaction
+was received via RPC, it is gossiped to all connected peers. Transactions received from other peers
+are forwarded to all the peers which have not already sent this particular transaction.
+
+This is enabled by CometBFT keeping the ID of each peer that has sent a transaction.
+This does not prevent a node to send the transaction to a node that has already
+received this transactions from its peers, thus creating a lot of duplicates. 
+
+This is the exact problem DOG was designed to tackle.
+
+### CAT protocol
+
+[Content-Addressable Transaction (CAT)](https://github.com/celestiaorg/celestia-core/blob/feature/cat/docs/celestia-architecture/adr-009-cat-pool.md) is a "push-pull" gossip protocol originally implemented by Celestia on top of the priority-mempool (aka v1), which existed in Tendermint Core until v0.36 and CometBFT until v0.37. 
+
+With CAT, nodes forward ("push") transactions received from users via RPC endpoints to their peers. When a node `A` receives a transaction from another node, it will notify its peers that it has the transaction with a `SeenTx(hash(tx))` message. A node `B`, upon receiving the `SeenTx` message from `A`, will check whether the transaction is in its mempool. If not, it will "pull" the transaction from `A` by sending a `WantTx(hash(tx))` message, to which `A` responds with a `Tx` message carrying the full transaction. While `SeenTx` and `HaveTx` messages are much smaller than `Tx` messages, these additional communication steps introduce latency when disseminating transactions. The CAT protocol was especially designed for networks with low throughput and large transaction sizes. 
+
+Efforts to port the CAT mempool to CometBFT were documented in [\#2027]. Experimental results on a small testnet from [\#1472] showed that CAT effectively reduces bandwidth usage. However, its impact on latency was not evaluated in those tests. Porting CAT was finally deprioritised in favor of DOG.
+
+### Limiting the number of peers a transaction is forwarded to
+
+CometBFT allows operators to configure `p2p.experimental_max_gossip_connections_to_non_persistent_peers` and `p2p.experimental_max_gossip_connections_to_persistent_peers`  as a maximum number of peers to send  transactions to ([#1558](https://github.com/cometbft/cometbft/pull/1558) and [#1584](https://github.com/cometbft/cometbft/pull/1584)). 
+This reduces bandwitdth compared to when they are disabled but there is no rule to determine which peers transactions
+are not forwarded to. DOG can be looked at as an enahanced, more informed version of this protocol.
+
+A node can use either this or DOG, but not both at the same time.
```

Comment:
Having both is confusing for users imho, and we should remove the former once DOG is out.

Reply:
I agree to remove the experimental feature.